src/partition/repart.c

```diff
@@ -6349,6 +6353,111 @@ static int parse_partition_types(const char *p, GptPartitionType **partitions, s
+                        NULL);
+        if (!store)
+                return log_debug_errno(
+                                SYNTHETIC_ERRNO(EINVAL),
+                                "Failed to open OpenSSL store via '%s': %s",
+                                private_key_path,
+                                ERR_error_string(ERR_get_error(), NULL));
+
+        _cleanup_(OSSL_STORE_INFO_freep) OSSL_STORE_INFO *info = OSSL_STORE_load(store);
+        if (!info && OSSL_STORE_error(store))
+                return log_debug_errno(
+                                SYNTHETIC_ERRNO(EINVAL),
+                                "Failed to load OpenSSL store via '%s': %s",
+                                private_key_path,
+                                ERR_error_string(ERR_get_error(), NULL));
+
+        EVP_PKEY_free(*private_key);
+        *private_key = OSSL_STORE_INFO_get1_PKEY(info);
+        if (!*private_key)
+                return log_debug_errno(
+                                SYNTHETIC_ERRNO(EINVAL),
+                                "Failed to load private key via '%s': %s",
+                                private_key_path,
+                                ERR_error_string(ERR_get_error(), NULL));
+
+        return 0;
+#else
+        return -EOPNOTSUPP;
+#endif
+}
+
+DISABLE_WARNING_DEPRECATED_DECLARATIONS
+DEFINE_TRIVIAL_CLEANUP_FUNC_FULL(ENGINE*, ENGINE_free, NULL);
+REENABLE_WARNING
+
+static int load_openssl_provider_legacy(
+                const char *signing_engine,
+                const char *private_key_path,
+                EVP_PKEY **private_key) {
+
+        assert(signing_engine);
+        assert(private_key_path);
+        assert(private_key);
+
+        DISABLE_WARNING_DEPRECATED_DECLARATIONS
+        _cleanup_(ENGINE_freep) ENGINE *e = ENGINE_by_id(signing_engine);
+        if (!e)
+                return log_debug_errno(
+                                SYNTHETIC_ERRNO(EINVAL),
+                                "Failed to load signing engine '%s': %s",
+                                signing_engine,
+                                ERR_error_string(ERR_get_error(), NULL));
+
+        if (ENGINE_init(e) == 0)
+                return log_debug_errno(
+                                SYNTHETIC_ERRNO(EINVAL),
+                                "Failed to initialize signing engine '%s': %s",
+                                signing_engine,
+                                ERR_error_string(ERR_get_error(), NULL));
+
+        EVP_PKEY_free(*private_key);
+        *private_key = ENGINE_load_private_key(e, private_key_path, NULL, NULL);
+        if (!*private_key)
+                return log_debug_errno(
+                                SYNTHETIC_ERRNO(EINVAL),
+                                "Failed to load private key from '%s': %s",
+                                private_key_path,
+                                ERR_error_string(ERR_get_error(), NULL));
+        REENABLE_WARNING
+
+        return 0;
+}
+
```

Comment:
hmm, this should probably be moved to openssl-util.[ch] right away, since we want to reuse that for systemd-measure i guess